shell/spy.go

```diff
@@ -1,19 +1,32 @@
 package shell
 
 import (
-	"bytes"
+	"fmt"
+	"io"
 	"log"
 	"os/exec"
+	"strings"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
-	var out bytes.Buffer
-	cmd.Stdout = &out
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult := strings.Split(cmd, " ")
+
+	if len(stringResult) == 1 {
+		return cmd, []string{}
+	}
+	return stringResult[0], stringResult[1:]
+}
+
+// Execute accepts a command argument and writes the result to a writer
+func Execute(writer io.Writer, commandLine string) {
+	name, args := ParseCommand(commandLine)
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = writer
 	err := cmd.Run()
 	if err != nil {
 		log.Fatal(err)
```

Comment:
We're not at home to Mr Log.Fatal! Libraries never panic (or terminate the program); that's a job for `main`. Instead, we return error.